presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java

```diff
@@ -248,6 +249,16 @@ public static void writeBigDecimal(DecimalType decimalType, BlockBuilder blockBu
         decimalType.writeSlice(blockBuilder, encodeScaledValue(value));
     }
 
+    public static BigDecimal rescale(BigDecimal value, DecimalType type)
+    {
+        value = value.setScale(type.getScale(), ROUND_UNNECESSARY);
```

Comment:
Indeed, it was previously `ROUND_HALF_UP`, but it was used after assertion that the scaled was increased, so there was no rounding (`ROUND_HALF_UP` was irrelevant).

I have changed it to `ROUND_UNNECESSARY` and removed scale increase assertion. This makes sure that the precision is not lost.